src/Authentication/AuthenticationServiceFactory.cs

```diff
@@ -0,0 +1,58 @@
+using System;
+using System.IO;
+using System.Threading.Tasks;
+using Azure.Core;
+using Azure.Identity;
+using Microsoft.Extensions.Options;
+using Microsoft.Graph.Cli.Configuration;
+using Microsoft.Graph.Cli.IO;
+using Microsoft.Graph.Cli.Utils;
+
+namespace Microsoft.Graph.Cli.Authentication;
+
+class AuthenticationServiceFactory {
+    public async Task<ILoginService> GetAuthenticationServiceAsync(AuthenticationStrategy strategy, string tenantId, string clientId) {
```

Comment:
in a production situation, we'll be using a public client (see powershell) by default, and won't have the tenant id. I just want to make sure we're not assuming it'll always be present in configuration files/be provided by the user.

Reply:
The tenant id can be null.